File: prompts/ios_technical_writter/mdToTxt.rb
require 'fileutils'

# Directories
markdown_dir = 'markdown'
text_dir = 'text'
output_file = 'templates.txt'

# Create text directory if it doesn't exist
Dir.mkdir(text_dir) unless Dir.exist?(text_dir)

# Clear the output file
File.open(output_file, 'w') {}

# Process each .md file
Dir.glob("#{markdown_dir}/*.md") do |md_file|
  filename = File.basename(md_file, '.md')
  txt_filename = "#{filename}.md.txt"
  txt_file_path = File.join(text_dir, txt_filename)

  # Read the content of the markdown file
  content = File.read(md_file)

  # Write the content to the .txt file in the text directory
  File.write(txt_file_path, content)

  # Append to the templates.txt file
  File.open(output_file, 'a') do |file|
    file.puts "# #{filename.capitalize} (from #{filename}.md)"
    file.puts content
    file.puts "\n"
  end
end

puts "Processing complete. Files saved in #{text_dir} and templates.txt created."

File: prompts/ios_technical_writter/mdToTxt.swift
import Foundation

let fileManager = FileManager.default
let markdownDir = "markdown"
let textDir = "text"
let outputFile = "templates.txt"

do {
    // Create text directory if it doesn't exist
    if !fileManager.fileExists(atPath: textDir) {
        try fileManager.createDirectory(atPath: textDir, withIntermediateDirectories: true)
    }

    // Clear the output file
    try "".write(toFile: outputFile, atomically: true, encoding: .utf8)

    // Process each .md file
    let markdownFiles = try fileManager.contentsOfDirectory(atPath: markdownDir)
    for mdFile in markdownFiles where mdFile.hasSuffix(".md") {
        let filename = (mdFile as NSString).deletingPathExtension
        let txtFilename = "\(filename).md.txt"
        let txtFilePath = "\(textDir)/\(txtFilename)"

        // Read the content of the markdown file
        let content = try String(contentsOfFile: "\(markdownDir)/\(mdFile)")

        // Write the content to the .txt file in the text directory
        try content.write(toFile: txtFilePath, atomically: true, encoding: .utf8)

        // Append to the templates.txt file
        if let fileHandle = FileHandle(forWritingAtPath: outputFile) {
            fileHandle.seekToEndOfFile()
            if let data = "# \(filename.capitalized) (from \(filename).md)\n\(content)\n\n".data(using: .utf8) {
                fileHandle.write(data)
            }
            fileHandle.closeFile()
        }
    }

    print("Processing complete. Files saved in \(textDir) and templates.txt created.")

} catch {
    print("An error occurred: \(error)")
}
